core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```diff
@@ -267,14 +275,20 @@ export class TopicEditorNavbarComponent {
     this.warningsAreShown = false;
     this.showTopicEditOptions = false;
     this.topic = this.topicEditorStateService.getTopic();
-    this.topicSkillIds = this.topic.getSkillIds();
+    if (this.topic) {
+      this.topicSkillIds = this.topic.getSkillIds();
+    }
     this.discardChangesButtonIsShown = false;
     this.validationIssues = [];
     this.prepublishValidationIssues = [];
     this.topicRights = this.topicEditorStateService.getTopicRights();
     this.directiveSubscriptions.add(
       this.undoRedoService.getUndoRedoChangeEventEmitter().subscribe(
-        () => this._validateTopic()
+        () => {
+          this.topic = this.topicEditorStateService.getTopic();
+          this.topicSkillIds = this.topic.getSkillIds();
```

Comment:
I think I would suggest getting rid of this.topicSkillIds and always using this.topic.getSkillIds() directly. That way you have a single source of truth and don't have to update multiple variables.

Reply:
Removed this.topicSkillIds from both this file and correspondingly from the spec.ts file